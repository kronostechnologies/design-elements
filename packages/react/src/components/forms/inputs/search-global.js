"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var search_input_1 = require("./search-input");
var SearchGlobal = function (_a) {
    var disabled = _a.disabled, id = _a.id, label = _a.label, onSearch = _a.onSearch;
    return (<search_input_1.SearchInput disabled={disabled} hasButton id={id} label={label} onSearch={onSearch}/>);
};
exports.SearchGlobal = SearchGlobal;
