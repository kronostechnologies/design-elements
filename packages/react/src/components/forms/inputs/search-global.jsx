import React from 'react';
import styled from 'styled-components';

import { SearchInput } from './search-input';
import { SearchButton } from '../../buttons/search-button.jsx';

const SearchSubmit = styled(SearchButton)`
  border-left: 0;
  border-radius: 0 0.25rem 0.25rem 0;
  position: relative;
`;

const SearchGlobal = ({ disabled, id, label, onSubmit }) => (
    <SearchInput
        disabled={disabled}
        id={id}
        label={label}
        onSubmit={onSubmit}
        global
    >
        <SearchSubmit
            disabled={disabled}
            type="submit"
            onClick={onSubmit}
        >
            Search
        </SearchSubmit>
    </SearchInput>
);

export { SearchGlobal };
