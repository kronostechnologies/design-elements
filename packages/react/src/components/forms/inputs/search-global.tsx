import React from 'react';

import { SearchInput, SearchInputProps } from './search-input';
const uuidv1 = require('uuid/v1');

const SearchGlobal = ({ disabled, label, searchCallback }: SearchInputProps) => {
    const id = uuidv1();
    return (
        <SearchInput
            disabled={disabled}
            hasButton
            id={id}
            label={label}
            searchCallback={searchCallback}
        />
    );
};

export { SearchGlobal };
