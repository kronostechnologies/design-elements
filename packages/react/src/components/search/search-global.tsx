import { forwardRef } from 'react';
import { CommonSearchProps, SearchInput } from './search-input';

export type SearchGlobalProps = CommonSearchProps

export const SearchGlobal = forwardRef<HTMLInputElement, SearchGlobalProps>(
    (props, ref) => (
        <SearchInput
            {...props /* eslint-disable-line react/jsx-props-no-spreading */}
            hasButton
            data-testid="search-input"
            ref={ref}
        />
    ),
);
