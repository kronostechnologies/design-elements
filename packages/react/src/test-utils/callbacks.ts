export function doNothing(): void {
    // do nothing
}
