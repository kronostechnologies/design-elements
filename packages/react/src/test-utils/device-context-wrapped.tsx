import React, { ReactElement } from 'react';
import { DeviceContextProvider, DeviceType } from '../components/device-context-provider/device-context-provider';

export function DeviceContextWrapped(children: ReactElement, device?: DeviceType): ReactElement {
    return (
        <DeviceContextProvider staticDevice={device}>
            {children}
        </DeviceContextProvider>
    );
}
