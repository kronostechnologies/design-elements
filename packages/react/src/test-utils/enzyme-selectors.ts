import { ReactWrapper, ShallowWrapper } from 'enzyme';

enum ModifierEnum {
    StartsWith = '^',
    EndsWith = '$',
    ContainsOptionalHyphen = '|',
    Contains = '*',
    ContainsWord = '~',
    None = '',
}
type SelectorModifier = `${ModifierEnum}`;

function filterHtmlNodesOnly<W extends ReactWrapper | ShallowWrapper>(
    nodes: W,
): W {
    return nodes
        .hostNodes()
        .filterWhere((x) => x.name() !== null && !x.name().startsWith('styled.')) as W;
}

export function findByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node over the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    const modifier = options?.modifier ?? '';

    let nodes = wrapper.find(`[data-testid${modifier}="${testId}"]`) as W;

    if (options?.nodeName) {
        nodes = nodes.filterWhere((x) => x.name() === options.nodeName) as W;
    }
    if (options?.htmlNodesOnly) {
        nodes = filterHtmlNodesOnly(nodes);
    }

    return nodes;
}

export function getByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node instead of the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    return findByTestId(wrapper, testId, options).at(0) as W;
}
