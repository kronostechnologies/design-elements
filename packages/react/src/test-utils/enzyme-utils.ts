import { ReactWrapper } from 'enzyme';
import { act } from 'react-dom/test-utils';

export async function waitForComponentToPaint<P = {}>(
    wrapper: ReactWrapper<P>,
    amount = 0,
): Promise<void> {
    await act(async () => {
        await new Promise((resolve) => setTimeout(resolve, amount));
        wrapper.update();
    });
}

export function expectFocusToBeOn(element: ReactWrapper): void {
    expect(document.activeElement).toBe(element.getDOMNode());
}
