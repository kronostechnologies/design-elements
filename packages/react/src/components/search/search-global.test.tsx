import { getByTestId } from '@design-elements/test-utils/enzyme-selectors';
import { shallow } from 'enzyme';
import React from 'react';
import { doNothing } from '../../test-utils/callbacks';
import { SearchGlobal } from './search-global';

jest.mock('@design-elements/utils/uuid');

describe('SearchGlobal', () => {
    it('should call onReset when search resets', () => {
        const onReset = jest.fn();
        const wrapper = shallow(<SearchGlobal onReset={onReset} />);

        getByTestId(wrapper, 'search-input').invoke('onReset')();

        expect(onReset).toHaveBeenCalledTimes(1);
    });

    it('should call onChange when search changes', () => {
        const onChange = jest.fn();
        const value = jest.fn();
        const event = jest.fn();
        const wrapper = shallow(<SearchGlobal onChange={onChange} />);

        getByTestId(wrapper, 'search-input').invoke('onChange')(value, event);

        expect(onChange).toHaveBeenCalledWith(value, event);
    });

    it('should call onSearch when search', () => {
        const onSearch = jest.fn();
        const value = jest.fn();
        const wrapper = shallow(<SearchGlobal onSearch={onSearch} />);

        getByTestId(wrapper, 'search-input').invoke('onSearch')(value);

        expect(onSearch).toHaveBeenCalledWith(value);
    });

    test('Matches the snapshot', () => {
        const wrapper = shallow(
            <SearchGlobal defaultValue="foo" label="Search" onSearch={doNothing} />,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
