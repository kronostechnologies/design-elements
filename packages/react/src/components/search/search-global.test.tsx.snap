// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchGlobal Matches the snapshot 1`] = `
<ForwardRef
  data-testid="search-input"
  defaultValue="foo"
  hasButton={true}
  label="Search"
  onSearch={[Function]}
/>
`;
